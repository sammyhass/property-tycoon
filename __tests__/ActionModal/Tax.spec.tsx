import ActionModalTax from '@/components/Game/ActionModal/Tax';
import { act } from '@testing-library/react';
import React from 'react';
import {
  fakeBoardSpace,
  fakeGameSettings,
  fakePlayer,
  fakePlayerState,
  renderWithGameContext,
} from '../fakers';

describe('Tax ActionModal', () => {
  test('matches snapshot', () => {
    const { container } = renderWithGameContext(<ActionModalTax />, {
      currentPlayer: fakePlayer('cat'),
      gameSettings: fakeGameSettings(),
      state: {
        boot: fakePlayerState(),
        cat: fakePlayerState(),
      },
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  test.skip('renders the correct text', () => {
    const { container } = renderWithGameContext(<ActionModalTax />, {
      currentPlayer: fakePlayer('cat'),
      gameSettings: fakeGameSettings({
        BoardSpaces: [
          fakeBoardSpace({
            board_position: 1,
            tax_cost: 200,
            space_type: 'TAX',
          }),
        ],
      }),
      state: {
        boot: fakePlayerState(),
        cat: fakePlayerState(),
      },
    });
    expect(container.textContent).toContain('Â£200 Tax');

    expect(container.textContent).toContain('Pay');
  });

  test('clicking button pays correct amount of tax to the bank', async () => {
    const payToFreeParking = jest.fn();
    const { container } = renderWithGameContext(<ActionModalTax />, {
      currentPlayer: fakePlayer('cat'),
      gameSettings: fakeGameSettings({
        BoardSpaces: [
          fakeBoardSpace({
            board_position: 1,
            tax_cost: 200,
            space_type: 'TAX',
          }),
        ],
      }),
      payToFreeParking,
      state: {
        boot: fakePlayerState(),
        cat: fakePlayerState({
          pos: 1,
        }),
      },
    });

    const button = container.querySelector('button');
    expect(button).toBeTruthy();
    await act(async () => {
      await button?.click();
      await new Promise(resolve => setTimeout(resolve, 1000));
    });
    expect(payToFreeParking).toHaveBeenCalledWith('cat', 200);
  });
});
