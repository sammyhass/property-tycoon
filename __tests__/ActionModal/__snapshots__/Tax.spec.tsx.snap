// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tax ActionModal matches snapshot 1`] = `
<div
  class="css-u6v8er"
>
  <h2
    class="chakra-heading css-0"
  >
    Pay the Tax
  </h2>
  <p
    class="chakra-text css-0"
  >
    You are paying 
    £0
     in taxes to free parking.
  </p>
  <p
    class="chakra-text css-1gjuw9l"
  >
    £0
  </p>
  <button
    class="chakra-button css-uxt1e8"
    type="button"
  >
    Pay
  </button>
</div>
`;
